### ISSUE TITLE
Show files in Folders for Cell Count folder and Create Video folder

### ISSUE THREAD

**Issue by EtalumaSupport** [EtalumaSupport/LumaViewPro#292]
it feels better to see the files you expect when choosing the folder to process.

**Comment by jmcoreymv** [comment]
I did some research on the library we're using for the folder selection and it doesn't allow for this.  I found several threads from other users trying to accomplish the same thing.  There is a dialog box that will show the files in the folder, but it then requires the user to click a file, which is not desired and awkward for this use-case.

I think we'd have to search out another UI library for this functionality, so it's not straightforward.

**Comment by ericweiner** [comment]
> **WRONG, corrected 2026-09-07.** The claim below that `IFileOpenDialog(FOS_PICKFOLDERS)` "does show files" is false: that dialog lists folders only, by Microsoft's definition, and Tk 8.6 already opens it. There is no one-site swap. See the correction comment of 2026-09-07 below. Kept for the record.

## Future home note

Still parked -- tkinter's askdirectory cannot show files. But the #750 fix (drafted on `fix/post-ship-2026-07` @ `1e5adf2e`) restructured dialog invocation so the Windows folder picker is now ONE function (`_platform_native_open_file`'s sibling `_platform_native_choose_folder` in `ui/file_dialogs.py`) behind one async runner. When this issue is picked up, an IFileOpenDialog(FOS_PICKFOLDERS) swap -- which does show files -- is a one-site change inside that primitive.

**Comment by ericweiner** [comment]
> **WRONG ROOT, corrected 2026-09-07.** The "legacy Browse for Folder" root below is false: Tk 8.6 already opens the modern pick-folders dialog on Vista and later, and that dialog lists folders only by design. The falsifier below could not tell the two dialogs apart. The scope widening, the labels and the native-picker ruling stand; the fix shape and the Enhance-cluster hold do not. Eric ruled 2026-09-07: Windows platform limitation, decide-not-to-fix. See the correction comment of 2026-09-07 below.

## Scope widened to EVERY folder picker; labels set; sequenced AHEAD of the Enhance cluster (Eric, 2026-09-03)

**Desired behaviour (Eric, 2026-09-03, verbatim):** *"on every file selection dialog, where normally on other programs you can see the files that are there, you can't see ANYTHING, it just shows an empty folder. for the capture folder, for selecting files for composite, stitching, Z stacks, etc. All of them."* -- and it must stay **native**: the standing ruling from #675 (*"The Kivy folder picker is absolutely horrible... revert to the native file picker"*) is reaffirmed today: *"we want native file choosers."*

So this issue is no longer "Cell Count and Create Video": it is **every folder-selection dialog on Windows**. That is the whole post-processing surface plus the capture folder.

**Root, as already on this thread and re-read at `67215b35`:** tkinter's `askdirectory` on Windows uses the legacy "Browse for Folder" dialog, which lists **folders only, never files**. Other applications use the modern `IFileOpenDialog` with `FOS_PICKFOLDERS`, which shows files greyed out so the user can tell which folder they are in. The 2023 comment was right that tkinter cannot do it; the 2024 "this is a kivy issue" was not -- the Kivy picker was the *attempted workaround* for this exact limitation, and #675 reverted it.

**The fix shape is already recorded here (2026-07-28) and is unchanged:** after #750 consolidated the Windows folder picker into ONE function, `ui/file_dialogs.py::_platform_native_choose_folder`, behind one async runner, swapping that primitive to `IFileOpenDialog(FOS_PICKFOLDERS)` is a **one-site change** that fixes every folder picker at once and stays native. Not planned here; triage ends at the root.

**Hypothesis status, stated honestly:** that the empty listing IS the legacy-dialog behaviour is the leading explanation and the one the code supports, but it has not been confirmed on a Windows machine today. **Falsifier, thirty seconds at any Windows unit:** open any folder picker -- if subfolders are visible and files are not, this root holds; if literally nothing is listed, something else is wrong in the path.

**Sequencing decision (Eric, 2026-09-03):** this issue is ahead of the Enhance cluster (#796 #815 #798). Every shape considered for that cluster hands the user off to this picker; fixing the Enhance chooser while the picker underneath hides files fixes nothing the user can feel.

**Links:** #675 (the Kivy attempt and its reversal -- the history of this defect); #796 #815 #798 (the Enhance chooser, downstream of this picker); #750 (the consolidation that made the one-site fix possible).

Labels: `Analyzed` (root and fix shape known), `High Priority` via `customer facing` -- every folder workflow on the platform most customers run.

**Comment by ericweiner** [comment]
> **WRONG, corrected the same day.** Eric's Friday answer (subfolders visible, files not) is consistent with BOTH the legacy tree and the modern pick-folders dialog, so it confirmed nothing about which dialog opens. The root recorded below is false and the disposition is superseded: decide-not-to-fix, Windows platform limitation. See the correction comment below.

## The falsifier is answered: the legacy-dialog root HOLDS (Eric, 2026-09-04)

The 2026-09-04 annotation above left the root as a hypothesis with a thirty-second falsifier: open any folder picker on a Windows unit -- subfolders visible and files not means the legacy "Browse for Folder" dialog; nothing listed at all means something else in the path.

Eric ran it on Friday 2026-09-04 on a Windows unit and answered **yes**: subfolders are visible, files are not. The answer was given in session but not recorded here or in the log; recorded now (2026-09-07) so the plan does not re-ask it.

**Root cause:** CONFIRMED -- tkinter's `askdirectory` on Windows opens the legacy folder-only dialog. No path defect.

**Disposition:** fix now, its own small plan (first in the post-autogain queue by Eric's 2026-09-03 sequencing): swap the one Windows folder-picker primitive in `ui/file_dialogs.py` to `IFileOpenDialog(FOS_PICKFOLDERS)`, native, every folder picker at once. Plan to follow; nothing built yet.

**Comment by ericweiner** [comment]
## CORRECTION: the "legacy dialog" root is wrong, the proposed fix is a no-op, and this is a Windows platform limitation (Eric, 2026-09-07: decide-not-to-fix)

The 2026-07-28 and 2026-09-04 annotations, and the one posted earlier today, all rested on one unchecked claim: that Windows's modern `IFileOpenDialog(FOS_PICKFOLDERS)` shows files greyed and that tkinter opens the legacy "Browse for Folder" tree instead. Both halves are false, checked today against the sources:

- **Tk already opens the modern dialog.** In Tk 8.6's `win/tkWinDialog.c`, `Tk_ChooseDirectoryObjCmd` takes the `GetFileNameVista` path whenever `VistaFileDialogsAvailable() && !ofnOpts.forceXPStyle`, and that path sets `FOS_PICKFOLDERS` for `OFN_DIR_CHOOSE`. The legacy `SHBrowseForFolder` tree appears only under the undocumented `-xpstyle` option, which LVP does not pass. Python 3.12's Tk 8.6.13 is on this path. The 2026-09-04 falsifier (subfolders visible, files not) could not tell the two dialogs apart, because both list folders only.
- **The pick-folders dialog lists folders only, by design.** Microsoft's definition of `FOS_PICKFOLDERS`: "Present an Open dialog that offers a choice of folders rather than files." No option shows files in it. VS Code, Windows Terminal and Chrome all behave this way.

**Desired behaviour (Eric, 2026-09-07):** the pickers stay native (the standing ruling from #675, reaffirmed 2026-09-03). On Windows, native and "see the files" conflict, and native wins.

**Root cause:** Windows has no native folder picker that lists files. LVP is already on the best native folder dialog Windows has. Not an LVP defect.

**Disposition:** decide-not-to-fix, a Windows platform limitation, ruled by Eric 2026-09-07 (*"mark it as a windows limitation and fix the comments"*). The alternatives put to him and not taken: pick a file inside the target folder through the open-file dialog (native, shows files, but a file click means a folder); a native picker followed by an in-app listing of what LVP found (new GUI surface); a non-native picker (rejected in #675). macOS is unaffected: its choose-folder panel lists files greyed.

**Code:** two comments in `ui/file_dialogs.py` claimed native pickers show file listings on every OS; corrected on `fix/292-picker-comments` (the linked commit), pending merge to `4.0.0-beta`. No behaviour change.

**Sequencing consequence:** the Enhance chooser cluster (#796 #815 #798) was held behind this issue because every shape hands off to this picker. That hold is void: the picker will not change. Its entry-point shape remains an open question for Eric.

Priority lowered to `Low Priority`: nothing is planned. The issue stays open for the reporter to close.

**Comment by ericweiner** [comment]
## Platform census and the one in-dialog option, for the record (2026-09-07)

Eric asked whether the other platforms behave the same and whether files could be listed beside the picker. Checked against the Tk source:

- **macOS: unaffected.** The choose-folder panel lists the folder's files greyed out.
- **Linux: the same limitation, and the dialog is Tk-drawn.** Tk does not use the desktop's picker on X11; `tk_chooseDirectory` is Tk's own directory chooser, and its listing (`tkfbox.tcl`, the `Update` procedure, class `TkChooseDir`) globs directories only. The code comments on `fix/292-picker-comments` now name Windows and Linux.
- **Windows, the only in-dialog option:** the modern dialog accepts custom controls through `IFileDialogCustomize` (static text, combo boxes, buttons, check boxes) and raises a folder-change event through `IFileDialogEvents::OnFolderChange`. No list control is available, so a filename pane is impossible; a **static text line updated on every folder change** ("24 TIFF images, 3 subfolders") is possible. Cost: Tk exposes none of this, so LVP would open the Windows dialog itself over COM (pywin32 as a new bundled dependency, or hand-written ctypes vtables and an event sink), roughly 100-150 lines plus a Windows bench. Linux gains nothing from it.

**Ruling stands (Eric, 2026-09-07):** platform limitation, decide-not-to-fix; none of the alternatives taken. Documented here so the next reader does not re-derive them.

**Comment by ericweiner** [comment]
## Branch state: comment-only correction drafted, pending merge

`fix/292-picker-comments` (off `4.0.0-beta` @ `132d09a9` = beta35), pushed, **not merged**:

- `df3d8e1e` -- the two comments in `ui/file_dialogs.py` stop claiming native pickers show files on every OS.
- `29c87b10` -- the same comments name Linux beside Windows (Tk draws its own directory chooser on X11 and globs directories only).

No behaviour change; no test changes. The three tests that read this file's source pass at the tip. Merges with any beta batch; nothing waits on it.

**Label:** left `Analyzed` / `Low Priority` -- not `Fixed Needs verification`, because nothing was fixed. The disposition is decide-not-to-fix (platform limitation) and the issue stays open for the reporter.

**Comment by ericweiner** [comment]
**Comment-only correction merged to `4.0.0-beta` (beta36). The disposition is unchanged: decide-not-to-fix, platform limitation.**

No behaviour changed. What shipped is the correction of three source comments that claimed the native pickers show files on every platform; they now name Windows and Linux as folders-only. The ruling recorded on 2026-09-07 stands, and the in-dialog option that does exist on Windows -- a static text line updated on `IFileDialogEvents::OnFolderChange` -- remains recorded and not taken.

Status: no fix pending; this issue stays open only as the record of the limitation, and closing it is yours whenever you want.
